pages/en/lb4/Testing-Your-Extensions.md

```diff
@@ -0,0 +1,305 @@
+---
+lang: en
+title: 'Testing your extension'
+keywords: LoopBack 4.0, LoopBack 4
+tags:
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Testing-your-extension.html
+summary:
+---
+
+## Overview
+
+An extension for LoopBack 4 is likely to be used by thousands and as such it's
+important to have a thorough test suite for an extension because:
+- Ensure your extension works correctly
+- Prevent regressions when new features are added or bugs are fixed
+- Helps demonstrate usage
+
+## Project Setup
+
+If using `@loopback/cli` to create a new extension project, it will come with a
+`test` folder with a recommended folder for various types of tests.
+An automated test suite requires a test runner to execute the tests and we
+recommend `mocha`. It is enabled by default if `@loopback/cli` was used to
+create the extension project. This will install and configure `mocha` as well
+as create a `test` command in your `package.json`.
+
+Assertion libraries such as [ShouldJS](http://shouldjs.github.io/) (as `expect`),
+[SinonJS](http://sinonjs.org/), and a swagger validator are made available
+through the convenitent `@loopback/testlab` package. This is installed if the
+project was created using the CLI.
+
+### Manual Setup - Using Mocha
+- Install `mocha` by running `npm i mocha`. This will save the package in
```

Comment:
`npm i --save-dev mocha` here to save it as a dev dep?

Reply:
^